08.ls_object/lib/long_formatter.rb

```diff
@@ -7,32 +7,31 @@
 class LongFormatter
   def initialize(file_paths)
     @file_paths = file_paths
-    file_stats = file_paths.map { |f| File::Stat.new(f) }
-    @max_file_path_count_nlink = file_stats.map { |f| f.nlink.to_s.size }.max
-    @max_file_path_count_owner_name = file_stats.map { |f| Etc.getpwuid(f.uid).name.size }.max
-    @max_file_path_count_group_name = file_stats.map { |f| Etc.getgrgid(f.gid).name.size }.max
-    @max_file_path_count_build_size = file_stats.map { |f| f.size.to_s.size }.max
-    @max_file_path_count_mtime = file_stats.map { |f| f.mtime.strftime('%-m %d %H:%M').size }.max
   end
 
   def run
-    file_stats = @file_paths.map { |file_path| LsFile.new(file_path) }
-    total_block = file_stats.sum(&:calc_block)
-    long_text = file_stats.map { |file_stat| build_detail_data(file_stat).join(' ') }
+    ls_files = @file_paths.map { |file_path| LsFile.new(file_path) }
+    max_file_path_count = {}
+    max_file_path_count[:nlink] = ls_files.map { |f| f.nlink.to_s.size }.max
+    max_file_path_count[:owner_name] = ls_files.map { |f| f.owner_name.size }.max
+    max_file_path_count[:group_name] = ls_files.map { |f| f.group_name.size }.max
+    max_file_path_count[:size] = ls_files.map { |f| f.size.to_s.size }.max
```

Comment:
ちょっとごちゃっとするのでメソッドに分けた方がスッキリしそうです。

```suggestion
    max_file_path_count = hogehoge(ls_files)
```

あと変数名がcountで終わると数値っぽく見えるので、`max_count_table`みたいな名前がいいかも。
ハッシュはtableと呼ばれることもあります。

Reply:
変数名を変更しmax_count_tableの計算用のメソッドを作成しました。